### PR TITLE
[csharp-netcore] Upgrade to System.Text.Json

### PR DESCRIPTION
**Pull request by devhl-labs**
Upgrading the generic host json library to System.Text.Json. Tested with anyOf, allOf, and oneOf schemas located at 
`modules\openapi-generator\src\test\resources\3_0`

### anyOf
```cs
string appleJson = "{\"color\": \"red\", \"kind\": \"apple\"}";

Fruit? apple = System.Text.Json.JsonSerializer.Deserialize<Fruit>(appleJson, Org.OpenAPITools.Client.ClientUtils.JsonSerializerOptions);
```
### allOf
```cs
string childJson = "{\"boosterSeat\": true, \"age\": 1, \"$_type\": \"Child\", \"lastName\": \"Smith\", \"firstName\": \"Abe\"}";

Child? child = System.Text.Json.JsonSerializer.Deserialize<Child>(childJson, Org.OpenAPITools.Client.ClientUtils.JsonSerializerOptions);

string adultJson = $"{{\"children\": [{childJson}], \"$_type\": \"Adult\", \"lastName\": \"Smith\", \"firstName\": \"Bob\"}}";

Adult? adult = System.Text.Json.JsonSerializer.Deserialize<Adult>(adultJson, Org.OpenAPITools.Client.ClientUtils.JsonSerializerOptions);
```
### oneOf
```cs
string appleJson = "{\"color\": \"red\", \"kind\": \"apple\"}";

Fruit? apple = System.Text.Json.JsonSerializer.Deserialize<Fruit>(appleJson, Org.OpenAPITools.Client.ClientUtils.JsonSerializerOptions);

string bananaJson = "{\"color\": \"red\", \"count\": 1}";

Fruit? banana = System.Text.Json.JsonSerializer.Deserialize<Fruit>(bananaJson, Org.OpenAPITools.Client.ClientUtils.JsonSerializerOptions);
```

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@mandrean (2017/08) @frankyjuang (2019/09) @shibayan (2020/02) @Blackclaws (2021/03) @lucamazzanti (2021/05)

### CODE REVIEW

**Comment by wing328** [comment]
Thanks for the PR. There some build failures:

https://github.com/OpenAPITools/openapi-generator/runs/5353892524?check_suite_focus=true
https://github.com/OpenAPITools/openapi-generator/runs/5353892549?check_suite_focus=true

Please take a look when you've time.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CSharpNetCoreClientCodegen.java [hunk 1]
```diff
@@ -1327,11 +1335,126 @@ public String toInstantiationType(Schema schema) {
                 cm.isNullable = true;
                 cm.anyOf.remove("Null");
             }
+
+            for (CodegenProperty cp : cm.readWriteVars) {
```
[the remark this answers is not in the document]
Comment (devhl-labs) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CSharpNetCoreClientCodegen.java:1339]: Yes, these two loops are fixing a bug in the default code gen, plus the EnsureInheritedVariablesArePresent method is also fixing a bug.

[the remark this answers is not in the document]
Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CSharpNetCoreClientCodegen.java:1339]: Do you mind submitting a PR to fix these bugs (ideally with tests) in the default codegen instead?

[the remark this answers is not in the document]
Comment (devhl-labs) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CSharpNetCoreClientCodegen.java:1339]: I like that to be a different PR. I'm trying to use this in a project.

[the remark this answers is not in the document]
Comment (devhl-labs) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CSharpNetCoreClientCodegen.java:1339]: Created a task for each of the bugs, and provided the url to those tasks in the code where I work around the issues. We should fix the source, but for now it is mitigated.

**Comment by devhl-labs** [comment]
I'd love to know how that commit means samples are not up to date. Something is broken there.

**Comment by wing328** [comment]
That's fixed in the master. It happens when we merge an old PR into master in which some new tests have been added to the test spec.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CSharpNetCoreClientCodegen.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CSharpNetCoreClientCodegen.java:1339]: 👌

**Pull request merged**